js/app/interfaces/card.js

```diff
@@ -283,6 +275,69 @@ const Card = new Lang.Interface({
                 widget.add(second_tag);
             }
         }
+        return widget;
+    },
+
+    // O promises, where art thou?
+    _count_set: function (set_obj, callback) {
+        let query = new QueryObject.QueryObject({
+            tags_match_any: set_obj.child_tags,
+            limit: -1,
+        });
+        Engine.get_default().get_objects_by_query(query, null, (engine, task) => {
+            let results, info;
+            try {
+                [results, info] = engine.get_objects_by_query_finish(task);
+                let reached_bottom = true;
+                results.forEach((obj) => {
+                    if (obj instanceof SetObjectModel.SetObjectModel) {
+                        reached_bottom = false;
+                        this._count_set(obj, callback);
+                    } else {
+                        this._child_count += 1;
+                    }
+                });
+                if (reached_bottom) {
+                    callback(this._child_count);
+                }
+            } catch (e) {
+                logError(e, 'Failed to load content from engine');
+                results = [];
+                info = { more_results: null };
+                return;
+            }
+        });
+    },
+
+    _create_inventory_widget: function () {
+        this._child_count = 0;
+        let label = new FormattableLabel.FormattableLabel();
+
+        this._count_set(this.model, (count) => {
+            label.label = count + ' lessons';
```

Comment:
> "Lessons" seems overly specific for a context label on all SetObjectModels. Is there a better way we could identify a set of lessons??

Not really - I dont want to make a separate LessonObject I think. I can make this text customizable though